comms/src/builder/builder_next.rs

```diff
@@ -0,0 +1,446 @@
+        let connection_manager = self.make_connection_manager(
+            executor.clone(),
+            node_identity.clone(),
+            peer_manager.clone(),
+            protocols,
+            conn_man_rx,
+            connection_manager_event_tx.clone(),
+        );
+
+        Ok(BuiltCommsNode {
+            executor,
+            connection_manager,
+            connection_manager_requester,
+            connection_manager_event_tx,
+            messaging_request_tx,
+            messaging_pipeline,
+            messaging,
+            messaging_event_tx,
+            inbound_message_rx,
+            node_identity,
+            peer_manager,
+            shutdown: self.shutdown,
+        })
+    }
+
+    pub async fn spawn(self) -> Result<CommsNode, CommsBuilderError> {
+        let BuiltCommsNode {
+            connection_manager,
+            connection_manager_requester,
+            connection_manager_event_tx,
+            messaging_request_tx,
+            messaging_pipeline,
+            inbound_message_rx,
+            executor,
+            node_identity,
+            shutdown,
+            peer_manager,
+            messaging,
+            messaging_event_tx,
+        } = self.build()?;
+
+        executor.spawn(connection_manager.run());
+
+        // Spawn messaging protocol
+        executor.spawn(messaging.run());
+
+        // Spawn inbound pipeline
+        let bounded_executor = BoundedExecutor::new(executor.clone(), messaging_pipeline.max_concurrent_inbound_tasks);
+        let inbound = pipeline::Inbound::new(bounded_executor, inbound_message_rx, messaging_pipeline.inbound);
+        executor.spawn(inbound.run());
+
+        // Spawn outbound pipeline
+        let outbound = pipeline::Outbound::new(executor.clone(), messaging_pipeline.outbound, messaging_request_tx);
+        executor.spawn(outbound.run());
+
+        Ok(CommsNode {
+            shutdown,
+            connection_manager_event_tx,
+            connection_manager_requester,
+            node_identity,
+            peer_manager,
+            messaging_event_tx,
+            executor,
+        })
+    }
+}
+
+/// Contains the built comms services
+struct BuiltCommsNode<TTransport, TInPipe, TOutPipe, TOutReq> {
+    connection_manager: ConnectionManager<TTransport, BoxedBackoff>,
+    connection_manager_requester: ConnectionManagerRequester,
+    connection_manager_event_tx: broadcast::Sender<Arc<ConnectionManagerEvent>>,
+    messaging_pipeline: pipeline::Config<TInPipe, TOutPipe, TOutReq>,
+    executor: runtime::Handle,
+    node_identity: Arc<NodeIdentity>,
+    messaging: MessagingProtocol,
+    messaging_event_tx: messaging::MessagingEventSender,
+    inbound_message_rx: mpsc::Receiver<InboundMessage>,
+
+    messaging_request_tx: mpsc::Sender<messaging::MessagingRequest>,
+    shutdown: Shutdown,
+    peer_manager: Arc<PeerManager>,
+}
+
+pub struct CommsNode {
```

Comment:
Nit: Maybe just a comment that this is essentially a handle?

Reply:
I'll add some comments